src/core/tests/pattern_output_index.cpp

```diff
@@ -0,0 +1,205 @@
+    auto input = std::make_shared<op::v0::Parameter>(element::f32, Shape{12, 4});
+    auto axis = op::v0::Constant::create(element::i32, Shape{}, {0});
+    auto split = std::make_shared<op::v1::Split>(input, axis, 3);
+
+    auto relu0 = std::make_shared<op::v0::Relu>(split->output(0));
+    auto relu1 = std::make_shared<op::v0::Relu>(split->output(1));
+    auto relu2 = std::make_shared<op::v0::Relu>(split->output(2));
+
+    // Pattern that can match any output using Or - this is how to achieve flexible matching
+    auto pattern_split = pattern::wrap_type<op::v1::Split>();
+    pattern_split->set_output_size(3);  // Pattern node needs to know it has 3 outputs
+    auto any_output = std::make_shared<pattern::op::Or>(
+        OutputVector{pattern_split->output(0), pattern_split->output(1), pattern_split->output(2)});
+    auto pattern_relu = pattern::wrap_type<op::v0::Relu>({any_output});
+
+    pattern::Matcher matcher(pattern_relu);
+
+    // Should match all outputs when using Or
+    EXPECT_TRUE(matcher.match(relu0->output(0)));
+    EXPECT_TRUE(matcher.match(relu1->output(0)));
+    EXPECT_TRUE(matcher.match(relu2->output(0)));
+}
+
+TEST(pattern_output_index, single_output_nodes_unaffected) {
+    // Test that single-output nodes are unaffected by strict index checking
+    auto input = std::make_shared<op::v0::Parameter>(element::f32, Shape{10});
+    auto relu = std::make_shared<op::v0::Relu>(input);
+    auto add_const = op::v0::Constant::create(element::f32, Shape{10}, {1.0f});
+    auto add = std::make_shared<op::v1::Add>(relu, add_const);
+
+    // Pattern for single-output nodes
+    auto pattern_relu = pattern::wrap_type<op::v0::Relu>();
+    auto pattern_add = pattern::wrap_type<op::v1::Add>({pattern_relu, pattern::any_input()});
+
+    pattern::Matcher matcher(pattern_add);
+    EXPECT_TRUE(matcher.match(add->output(0)));  // Should match normally
+}
+
+TEST(pattern_output_index, verify_index_mismatch_prevents_matching) {
+    // Comprehensive test to verify that index mismatches prevent matching
+    // With the new behavior, strict checking is automatic
+    auto input = std::make_shared<op::v0::Parameter>(element::f32, Shape{12});
+    auto axis = op::v0::Constant::create(element::i32, Shape{}, {0});
+    auto split_lengths = op::v0::Constant::create(element::i32, Shape{4}, {3, 3, 3, 3});
+    auto split = std::make_shared<op::v1::VariadicSplit>(input, axis, split_lengths);
+
+    // Create operations on all outputs
+    std::vector<std::shared_ptr<op::v0::Relu>> relus;
+    for (size_t i = 0; i < 4; ++i) {
+        relus.push_back(std::make_shared<op::v0::Relu>(split->output(i)));
+    }
+
+    // Test each pattern output against all graph outputs
+    for (size_t pattern_idx = 0; pattern_idx < 4; ++pattern_idx) {
+        auto pattern_split = pattern::wrap_type<op::v1::VariadicSplit>();
+        pattern_split->set_output_size(4);  // Pattern node needs to know it has 4 outputs
```

Comment:
```suggestion
        auto pattern_split = pattern::wrap_type<op::v1::VariadicSplit>();
        pattern_split->set_output_size(split_lengths);  // Pattern node needs to know it has 4 outputs
```
Can number of splits be set in `pattern::wrap_type<op::v1::VariadicSplit>();`?